packages/vitest/src/node/transform.ts

```diff
@@ -18,28 +18,40 @@ export async function transformRequest(ctx: Vitest, id: string) {
   return promiseMap.get(id)
 }
 
+function getTransformMode(ctx: Vitest, id: string) {
+  const withoutQuery = id.split('?')[0]
```

Comment:
why without query?

Reply:
Why with query then?

I think the original approach is a bit problematic as I didn't consider the query, as `.vue` should be treated as client transform but `.vue?vue&lang.ts` be treated as ssr transform is weird.

Also, a query could be like `.ts?v=xxx` that we should generally not consider them I think

Reply:
can it be a `?v=` tho? I think by default we should check withour query, but give authors flexiblity? 
but maybe I'm thinking too much into it.

Reply:
can't really seem to find a proper usecase where it would be needed to have a query